client/components/test-mode-notice/index.js

```diff
@@ -103,15 +103,16 @@ export const getNoticeMessage = ( topic ) => {
 };
 
 export const TestModeNotice = ( { topic } ) => {
-	return (
-		isInTestMode() && (
+	if ( isInTestMode() ) {
+		return (
 			<Notice
 				className="wcpay-test-mode-notice"
 				status="warning"
 				isDismissible={ false }
 			>
 				{ getNoticeMessage( topic ) }
 			</Notice>
-		)
-	);
+		);
+	}
+	return null;
```

Comment:
I just noticed this isn't a a `.tsx` file, but let's go with `undefined` anyway. It'll just be of more help when we eventually migrate this file 🙂

Reply:
Fixed in 58c7d0a8d482135e99d4ece5efb0a4bb044e1422

Reply:
Turns out it's actually better to just return an empty component `<></>` since the type system doesn't like `undefined`. Made that change in ce26e93c21448b01b91518590f826adc1870551a 😅